MedallionShell/Shell.cs

```diff
@@ -76,6 +83,24 @@ public Command Run(string executable, IEnumerable<object>? arguments = null, Act
             return command;
         }
 
+        // internal for testing
+        internal static string? GetFullPathUsingSystemPathOrDefault(string executable)
+        {
+            var paths = Environment.GetEnvironmentVariable("PATH")!.Split(Path.PathSeparator);
```

Comment:
It's okay to use `!` here because both Windows and Unix-based OSes use `PATH` per https://en.wikipedia.org/wiki/PATH_(variable).

Reply:
That doesn't guarantee that this environment variable is defined for a particular process, though.

We should handle the case where it isn't (should behave as if the path were empty...)